### PR TITLE
Conversation display participants

### PR DESCRIPTION
**Pull request by PopDaph**
This PR adds a `participants` array to `ConversationType`, which is filled  on `getConversation`.

Then a few changes on the components `Conversation` and `ConversationTitle`: 
- `Conversation` is in charge of listening to the streamed events, and it calls mutate if there's a change. (No more logic with onTitleUpdate)
- `ConversationTitle` now loads the conversation with `useConversation`, meaning it will get automatically refreshed if any other component has used mutate. 

I want to rename `ConversationTitle` to `ConversationHeader` but I plan on doing it in a separate PR to not mess with the diff. 

Participants are displayed in the header: 
Capture d’écran 2023-09-26 à 14 24 49

### CODE REVIEW

File: front/types/assistant/conversation.ts [hunk 1]
```diff
@@ -126,6 +126,7 @@ export type ConversationType = {
   title: string | null;
   visibility: ConversationVisibility;
   content: (UserMessageType[] | AgentMessageType[])[];
+  participants: { id: string; name: string; pictureUrl: string | null }[];
```
Comment (spolu) [inline front/types/assistant/conversation.ts:129]: `id` is unclear and a client-side concern

I think we should have a UserParticipant and an AgentParticipant with specific fields I guess configurationId handle and pictureUrl. User can have specific fields about the user as appropriate

Then from there you'll be able to set a unique key client side

**Review by spolu — CHANGES_REQUESTED** [review]
We def def def def want superposed picture URL with humans on one side and agents on the other.

Left a request change on the signature of participants 

Have you checked that there is only one HTTP call to getConversation when you load the page?

File: front/components/assistant/conversation/ConversationTitle.tsx [hunk 2]
```diff
@@ -27,13 +33,32 @@ export function ConversationTitle({
...
+
   return (
     <div className="grid h-full max-w-full grid-cols-[1fr,auto] items-center gap-4">
       <div className="overflow-hidden truncate">
-        <span className="font-bold">{title}</span>
+        <span className="font-bold">{conversation?.title || ""}</span>
       </div>
 
-      <div className="flex space-x-1">
+      <div className="flex gap-1">
```
Comment (spolu) [inline front/components/assistant/conversation/ConversationTitle.tsx:52]: likely overflows on mobile ?

File: front/types/assistant/conversation.ts (hunk 1 above)
Comment (PopDaph) [reply front/types/assistant/conversation.ts]: I added a commit!

File: front/components/assistant/conversation/ConversationTitle.tsx (hunk 2 above)
Comment (PopDaph) [reply front/components/assistant/conversation/ConversationTitle.tsx]: Here's a mobile screenshot, ok for you?

Capture d’écran 2023-09-26 à 14 17 10

File: front/types/assistant/conversation.ts [hunk 3]
```diff
@@ -126,6 +126,21 @@ export type ConversationType = {
...
   visibility: ConversationVisibility;
   content: (UserMessageType[] | AgentMessageType[])[];
+  participants: {
+    users: UserParticipant[];
+    agents: AgentParticipant[];
+  };
+};
+
+export type UserParticipant = {
+  userName: string;
```
Comment (spolu) [inline front/types/assistant/conversation.ts:136]: to be consistent with UserMessageContext type this should be `username` and `fullName`

File: front/components/assistant/conversation/ConversationTitle.tsx (hunk 2 above)
Comment (PopDaph) [reply front/components/assistant/conversation/ConversationTitle.tsx]: I hid them on mobile, and increased a bit the stacking: 

Capture d’écran 2023-09-26 à 14 24 49

**Review by spolu — CHANGES_REQUESTED** [review]

**Comment by spolu** [comment]
LGTM otherwise \o/ assuming we confirm we're making only one API call

File: front/types/assistant/conversation.ts (hunk 3 above)
Comment (PopDaph) [reply front/types/assistant/conversation.ts]: I added a commit!

**Pull request merged**